Update randomness doc (#1281)
onflow/docs: docs/tools/kit/index.md

#### Notes:

> * Randomness is generated using Flow’s **on-chain `revertibleRandom`**, producing pseudorandom values tied to block and script execution.

Author: there are no transactions in this case, it's all script based (under the hood, this makes a difference in how randomness works).

> * Values are **deterministic**: The values returned for identical calls within the same block will be identical.

Author: What is described here in determinism, so I moved the "deterministic" note here

> * If `count ` is larger than one, the returned values are disctinct.

Author: added this so that developers don't think determinism means all `count` values are equal

> * If `count ` is larger than one, the returned values are disctinct.

Reviewer: ```suggestion
* If `count ` is larger than one, the returned values are distinct.
```

> * If `count ` is larger than one, the returned values are disctinct.

Reviewer 2: ```suggestion
* If `count ` is larger than one, the returned values are distinct.
```

> * This hook is designed for simple use cases that don't require unpredictability, such as randomized UIs.

Author: commit-reveal solves the reversibility issue, but here in the script nothing is reversible so commit-reveal doesn't add any advantage. 

The main thing developers should be aware of is that the block random source in the lib is publicly known (since this is script-based). There is nothing wrong with that in many applications like the example mentioned here (UIs)

>   Since the hook uses script executions on existing blocks, Flow's randon source is already public and the randoms are predictable. 

Reviewer 3: ```suggestion
  Since the hook uses script executions on existing blocks, the random source is already public and the randoms are predictable. 
```

> * For **more advanced use cases** that **do** require on-chain randomness logic via transactions, Flow provides built-in support using Cadence's `revertibleRandom` and [commit-reveal scheme].

Author: The more advanced cases are the ones with the logic fully on-chain via transactions (not scripts). This is where randomness is unpredictable, and reversibility can or cannot be an issue, so I added the mention of transactions here. Both "reversibleRandom" and "commit-reveal" work in that case depending on the app.
